mne.datasets.utils ImportError

Locally I hit an `ImportError` If I invoke a single unit test or test module in MNE-BIDS, e.g.

`pytest mne_bids/tests/test_fileio.py -k test_open_lock_readonly_parent_falls_back`:

```python-traceback
mne_bids/tests/test_fileio.py E                                                                                                   [100%]

================================================================ ERRORS =================================================================
______________________________________ ERROR at setup of test_open_lock_readonly_parent_falls_back ______________________________________
mne_bids/conftest.py:29: in monkeypatch_mne
    mne.datasets.utils._MODULES_TO_ENSURE_DOWNLOAD_IS_FALSE_IN_TESTS = (
    ^^^^^^^^^^^^^^^^^^
../../../miniforge3/envs/mnedev_311/lib/python3.11/site-packages/lazy_loader/__init__.py:94: in __getattr__
    raise AttributeError(f"No {package_name} attribute {name}")
E   AttributeError: No mne.datasets attribute utils
------------------------------ generated xml file: /Users/scotterik/devel/repos/mne-bids/junit-results.xml ------------------------------
========================================================= slowest 20 durations ==========================================================

(2 durations < 0.005s hidden.  Use -vv to show these durations.)
======================================================== short test summary info ========================================================
ERROR mne_bids/tests/test_fileio.py::test_open_lock_readonly_parent_falls_back - AttributeError: No mne.datasets attribute utils
```

The easiest fix on the MNE-BIDS side is probably to replace `mne.datasets.utils` accesses with:

```python
from mne.datasets import utils as ds_utils

# then ds_utils.some_function
```

And it may also be worth a fix upstream to MNE-Python if `mne.datasets.utils` is truly a public module. I assume this error originates from the MNE-Python Lazy Loader PR https://github.com/mne-tools/mne-python/pull/11838

--- a/mne_bids/conftest.py
+++ b/mne_bids/conftest.py
@@ -5,6 +5,7 @@
 from datetime import datetime
 
 import mne
+import mne.datasets.utils
 import pytest
 
 

